#BN-12 feat: updating user access

Charlesntwari opened the pull request:
# PR Description
This PR gives admin license to disable and immediately email user in case there is any rule violation occurred  
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] disable user account
- [x] emailing user after disable their account 
- [x] ......
# Description of tasks that were expected to be completed and that are not yet completed
- [x] none

# How this may be tested?
This can be tested by changing the status of user we have in our database
Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
3. open postman or swagger
4. update status of user
# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

Charlesntwari committed:
admin license to block user account

uwdjany reviewed (commented):
user email and password can be hidden using .env files.

Charlesntwari committed:
added on middleware

Charlesntwari commented:
> user email and password can be hidden using .env files.

thank you! it is now modified

Charlesntwari committed:
added on some translations

Honore123 reviewed (commented):
resolve conflicts

Charlesntwari committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into admin-license

Charlesntwari committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into admin-license

chrisostome01 commented on src/controllers/userconroller.js:
```diff
@@ -0,0 +1,91 @@
+    return res.status(500).json(error.message);
+  }
+});
+// eslint-disable-next-line consistent-return
```
Uncomment this line.

Charlesntwari replied:
solved

chrisostome01 commented on src/controllers/userconroller.js:
```diff
@@ -0,0 +1,91 @@
+      subject: `Your status has been updated to ${status}`,
+      text: `Dear ${user.firstName}, your status has been updated to ${status}.`,
+    });
+    console.log(`Email sent to ${user.email}`);
```
Please ensure that no consoles are left to be on production

Charlesntwari replied:
resolved

chrisostome01 commented on src/database/migrations/20230327061757-create-cart.js:
Please ignore this folder inside eslint

Charlesntwari replied:
solved

chrisostome01 commented on src/controllers/userconroller.js:
```diff
@@ -0,0 +1,91 @@
+const transporter = nodemailer.createTransport({
+  service: 'gmail',
+  auth: {
+    user: 'charlesntwari2@gmail.com',
+    pass: 'orvilqiccnltvonu',
```
Please put this inside the .env file. your email password should not be exposed like this

chrisostome01 commented on src/controllers/userconroller.js:
```diff
@@ -0,0 +1,91 @@
+
+const User = db.user;
+export const getAllUsers = catchAsync(async (req, res) => {
+  // console.log(User);
```
remove this comment

chrisostome01 commented on src/locales/en/translation.json:
`user not found` is duplicated couple of times

chrisostome01 commented on src/locales/fr/translation.json:
user not found is duplicated a couple of times

Charlesntwari committed:
fix (bug): resolve conversation
- ignoring migration folder inside eslint
- remove user_not_found duplication in translations
- remove all consoles
- add email information in .env
- uncomment eslints

chrisostome01 commented on src/locales/en/translation.json:
```diff
@@ -1,5 +1,11 @@
+  "failId": "no user found with that ID",
+  "retrievedAll": "all users retrieved",
+  "retrieved": "user retrieved",
+  "updateStatus": "status updated successfully",
```
- use snake case instead and remove fail because fail means a different thing. I would suggest casting the key to what the message says
- for fail you can reuse "user_not_found"
- here is an example of what you trying to do which is correct "user_not_found": "User not found",

chrisostome01 commented on src/locales/fr/translation.json:
```diff
@@ -1,5 +1,11 @@
+  "failId": "aucun utilisateur trouvé avec cet ID",
+  "retrievedAll": "tous les utilisateurs récupérés",
+  "retrieved": "utilisateur récupéré",
+  "updateStatus": "statut mis à jour avec succès",
```
- use snake case instead and remove fail because fail means a different thing. I would suggest casting the key to what the message says
- fail is not meaning full to the message you are responding.
- here is an example of what you trying to do which is correct "user_not_found": "utilisateur non trouvé".

chrisostome01 reviewed (approved):
# well done @Charlesntwari
